runtime/JavaScript/src/antlr4/Recognizer.d.ts

```diff
@@ -6,4 +6,7 @@ export declare class Recognizer<TSymbol> {
 
     removeErrorListeners(): void;
     addErrorListener(listener: ErrorListener<TSymbol>): void;
+    getErrorListenerDispatch(): ErrorListener<TSymbol>;
```

Comment:
Can we rename this to say `getRootErrorListener()` (open to suggestions). Appreciate it's an instance of `ErrorListenerDispatch` but that's a terrible name...

Reply:
The name `mode` is really not a good property name, because there is already a `mode()` method in the JavaScript Lexer.
Maybe a property for JavaScript/TypeScript called `lexerMode` or `tokenizerMode`?
And the `mode()` method would be depreciated in the JavaScript Lexer.

I don't like the name getErrorListenerDispatch either.

Reply:
Actually we can't change that name without breaking backwards compatibility in JS...

Reply:
Indeed, unfortunately the `getErrorListenerDispatch` method should not be renamed.
I think an independent TypeScript runtime (no `*.js`, no `*.d.ts`) would be ideal for this and other more modern designs.
Maybe in ANTLR5?

I remove my `setMode` method because there is already a `mode` method that does the same thing.
However, a `setMode`/`getMode` pair would have been more elegant than a `mode`/`getMode` pair.